apps/src/netsim/NetSimRouterLogModal.js

```diff
@@ -267,18 +455,24 @@ NetSimRouterLogModal.prototype.setShard = function (newShard) {
     this.eventKeys_.registeredWithShard = newShard;
   }
 
+  // When changing shards, reset log so we fetch the whole thing next time.
+  this.logEntries_.length = 0;
+  this.latestRowID_ = 0;
   this.shard_ = newShard;
 };
 
 /**
  * Handle log table changes.
- * @param {LogEntryRow[]} logRows
  * @private
  */
-NetSimRouterLogModal.prototype.onLogTableChange_ = function (logRows) {
+NetSimRouterLogModal.prototype.onLogTableChange_ = function () {
   var headerSpec = NetSimGlobals.getLevelConfig().routerExpectsPacketHeader;
-  this.logEntries_ = logRows.map(function (row) {
+  var newRows = this.shard_.logTable.readAllFromID(this.latestRowID_ + 1);
+  var newLogEntries = newRows.map(function (row) {
+    this.latestRowID_ = Math.max(row.id, this.latestRowID_);
```

Comment:
Are you sure that `readAllFromID` guarantees a sorted result?  It doesn't seem to, unless that's guaranteed when using `for (... in ...)` iteration on an object with numeric keys.